pkg/sass-parser/lib/src/expression/list.test.ts

```diff
@@ -0,0 +1,1517 @@
+            beforeEach(() => void (node = create()));
+
+            it('has sassType list', () => expect(node.sassType).toBe('list'));
+
+            it('has brackets', () => expect(node.brackets).toBe(true));
+
+            it('has an unknown separator', () =>
+              expect(node.separator).toBe(null));
+
+            it('has an element', () => {
+              expect(node.nodes).toHaveLength(1);
+              expect(node).toHaveStringExpression(0, 'foo');
+            });
+          });
+        }
+
+        describeNode('parsed', () => utils.parseExpression('[foo]'));
+
+        describeNode(
+          'constructed manually',
+          () =>
+            new ListExpression({
+              separator: null,
+              brackets: true,
+              nodes: [{text: 'foo'}],
+            }),
+        );
+
+        describeNode('constructed from ExpressionProps', () =>
+          utils.fromExpressionProps({
+            separator: null,
+            brackets: true,
+            nodes: [{text: 'foo'}],
+          }),
+        );
+      });
+
+      describe('with a comma separator', () => {
+        function describeNode(
+          description: string,
+          create: () => ListExpression,
+        ): void {
+          describe(description, () => {
+            beforeEach(() => void (node = create()));
+
+            it('has sassType list', () => expect(node.sassType).toBe('list'));
+
+            it('has brackets', () => expect(node.brackets).toBe(true));
+
+            it('has a comma separator', () => expect(node.separator).toBe(','));
+
+            it('has an element', () => {
+              expect(node.nodes).toHaveLength(1);
+              expect(node).toHaveStringExpression(0, 'foo');
+            });
+          });
+        }
+
+        describeNode('parsed', () => utils.parseExpression('[foo,]'));
+
+        describeNode(
+          'constructed manually',
+          () =>
+            new ListExpression({
+              separator: ',',
+              brackets: true,
+              nodes: [{text: 'foo'}],
+            }),
+        );
+
+        describeNode('constructed from ExpressionProps', () =>
+          utils.fromExpressionProps({
+            separator: ',',
+            brackets: true,
+            nodes: [{text: 'foo'}],
+          }),
+        );
+      });
+    });
+  });
+
+  describe('with multiple elements ', () => {
+    describe('without brackets', () => {
+      describe('with a space separator', () => {
+        function describeNode(
+          description: string,
+          create: () => ListExpression,
+        ): void {
+          describe(description, () => {
+            beforeEach(() => void (node = create()));
+
+            it('has sassType list', () => expect(node.sassType).toBe('list'));
+
+            it('has no brackets', () => expect(node.brackets).toBe(false));
+
+            it('has a comma separator', () => expect(node.separator).toBe(' '));
+
+            it('has an element', () => {
```

Comment:
Optional: Should this and the other tests with multiple elements say "has two elements" or something like that?